phoenixd/phoenixd.c

```diff
@@ -250,8 +250,8 @@ int main(int argc, char *argv[])
 	}
 
 	if (output) {
-		if (kernel == NULL || initrd == NULL) {
-			fprintf(stderr, "Output file needs kernel and initrd paths\n");
+		if (kernel == NULL) {
```

Comment:
If `initrd` is `NULL` it can cause fault in usb_imx.c:395 and usb_imx.c:432

Reply:
I'm not 100% sure, but it looks like a copy-paste mistake. In both cases, we are trying to open a file named `prog`, but printing error messages: `fprintf(stderr, "Can't open initrd file %s\n", initrd);` and `fprintf(stderr, "Could not open file %s\n", initrd);`. I think substituting `initrd` with `prog` should work and do, what the original implementer might have meant.

Reply:
I've made that fix in the next version.